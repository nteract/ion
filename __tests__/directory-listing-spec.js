// @flow
import * as React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import {
  Entry,
  Listing,
  Name,
  Icon,
  LastSaved,
  GroupedDirectoryListings
} from "../src";

describe("Listing", () => {
  it("accepts props and renders entries in directory in a table", () => {
    const component = (
      <Listing>
        <Entry>
          <Icon fileType={"directory"} />
          <Name>{"linky"}</Name>
          <LastSaved
            last_modified={
              "Fri Jun 22 2018 00:15:55 GMT-0400 (Eastern Daylight Time)"
            }
          />
        </Entry>
      </Listing>
    );
    expect(toJson(component)).toMatchSnapshot();
  });
});

describe("Entry", () => {
  it("accepts props and renders entries in directory", () => {
    const component = shallow(
      <Entry>
        <Icon fileType={"directory"} />
        <Name>{"linky jr"}</Name>
        <LastSaved
          last_modified={
            "Fri Jun 22 2018 00:15:55 GMT-0400 (Eastern Daylight Time)"
          }
        />
      </Entry>
    );
    expect(toJson(component)).toMatchSnapshot();
  });
});

describe("GroupedDirectoryListings", () => {
  it("accepts contents as props and renders a directory listing", () => {
    const component = shallow(
      <div>
        <GroupedDirectoryListings contents={contents} basepath={"/"} />
      </div>
    );
    expect(toJson(component)).toMatchSnapshot();
  });
});

const contents = [
  {
    type: "directory",
    mimetype: null,
    format: "json",
    content: null,
    writable: true,
    name: "photos",
    path: "#",
    created: "2018-07-25T02:02:32.242Z",
    last_modified: "2018-07-25T02:02:32.242Z"
  },
  {
    type: "notebook",
    mimetype: null,
    format: "json",
    content: null,
    writable: true,
    name: "CNN.ipynb",
    path: "#",
    created: "2018-07-25T02:02:32.242Z",
    last_modified: "2018-07-25T02:02:32.257Z"
  },
  {
    type: "file",
    mimetype: null,
    writable: true,
    format: "text",
    content: null,
    name: "node.js",
    path: "#",
    created: "2018-07-25T02:02:32.270Z",
    last_modified: "2018-07-25T02:02:32.270Z"
  }
];
