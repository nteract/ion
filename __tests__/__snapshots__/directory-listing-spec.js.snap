// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entry accepts props and renders entries in directory 1`] = `
<tr
  className="jsx-3511019751 directory-entry"
>
  <Icon
    className="directory-entry-field"
    color="#0366d6"
    fileType="directory"
    key=".0"
  />
  <Name
    className="directory-entry-field"
    key=".1"
  >
    linky jr
  </Name>
  <LastSaved
    className="directory-entry-field"
    key=".2"
    lastModified={2018-07-25T04:41:40.939Z}
    last_modified="Fri Jun 22 2018 00:15:55 GMT-0400 (Eastern Daylight Time)"
  />
  <JSXStyle
    css="tr.jsx-3511019751{border-top:1px solid #eaecef;}tr.jsx-3511019751:hover{background-color:#f6f8fa;-webkit-transition:background-color 0.1s ease-out;transition:background-color 0.1s ease-out;}tr.jsx-3511019751:first-child{border-top:none;}tr.jsx-3511019751:last-child{border-bottom:none;}"
    styleId="3511019751"
  />
</tr>
`;

exports[`GroupedDirectoryListings accepts contents as props and renders a directory listing 1`] = `
<div>
  <GroupedDirectoryListings
    basepath="/"
    contents={
      Array [
        Object {
          "content": null,
          "created": "2018-07-25T02:02:32.242Z",
          "format": "json",
          "last_modified": "2018-07-25T02:02:32.242Z",
          "mimetype": null,
          "name": "photos",
          "path": "#",
          "type": "directory",
          "writable": true,
        },
        Object {
          "content": null,
          "created": "2018-07-25T02:02:32.242Z",
          "format": "json",
          "last_modified": "2018-07-25T02:02:32.257Z",
          "mimetype": null,
          "name": "CNN.ipynb",
          "path": "#",
          "type": "notebook",
          "writable": true,
        },
        Object {
          "content": null,
          "created": "2018-07-25T02:02:32.270Z",
          "format": "text",
          "last_modified": "2018-07-25T02:02:32.270Z",
          "mimetype": null,
          "name": "node.js",
          "path": "#",
          "type": "file",
          "writable": true,
        },
      ]
    }
  />
</div>
`;

exports[`Listing accepts props and renders entries in directory in a table 1`] = `undefined`;
